dotfiles/run.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+DOTFILES_URI="${DOTFILES_URI}"
+DOTFILES_USER="${DOTFILES_USER}"
+
+if [ -n "$${DOTFILES_URI// }" ]; then
+  if [ -z "$DOTFILES_USER" ]; then
+    DOTFILES_USER="$USER"
+  fi
+
+  echo "✨ Applying dotfiles for user $DOTFILES_USER"
+
+  if [ "$DOTFILES_USER" = "$USER" ]; then
+    coder dotfiles "$DOTFILES_URI" -y 2>&1 | tee -a ~/.dotfiles.log
+  else
+    # The "eval echo ~$USER" part is used to dynamically get the home directory of the user, see https://superuser.com/a/484280
+    # sudo -u coder sh -c 'eval echo ~$USER' -> "/home/coder"                                                                                               ~/projects/coder-modules
+    # sudo sh -c 'eval echo ~$USER'          -> "/root"
+
+    sudo -u "$DOTFILES_USER" sh -c "$(which coder) dotfiles \"$DOTFILES_URI\" -y 2>&1 | tee -a $(eval echo ~$USER)/.dotfiles.log"
```

Comment:
The escaping here is going to be gnarly, as is it'll return the current users directory, not the target users:

```suggestion
    sudo -u "$DOTFILES_USER" sh -c "'$(which coder)' dotfiles '$DOTFILES_URI' -y 2>&1; echo tee \""'$(eval echo ~$USER)'"\"/.dotfiles.log"
```

Added safety `'`s as well to prevent splitting.

Reply:
When I wrote the script, I was using `$USER` instead of `$DOTFILES_USER` but quickly realized that was a mistake.
So I only forgot to replace the variable.

This is what I meant to do, do you see any issues with doing it this way ?
```
sudo -u root sh -c "'$(which coder)' dotfiles '$DOTFILES_URI' -y 2>&1 | tee '$(eval echo ~$DOTFILES_USER)/.dotfiles.log'"
```

---

```sh
coder@env> sudo -u root sh -c "'$(which coder)' dotfiles '$DOTFILES_URI' -y 2>&1 | tee '$(eval echo ~$DOTFILES_USER)/.dotfiles.log'"
Checking if dotfiles repository already exists...
Found dotfiles repository at /root/.config/coderv2/dotfiles
warning: redirecting to [...]
Already up to date.
Running setup.sh...
[+] Setting up dotfiles...
[+] Absolute path: /root/.config/coderv2/dotfiles
[+] Changing default shell...
[+] Changing PAM policy for chsh
[+] Setting default shell
[+] Reverting PAM policy for chsh
[+] Could not change default shell (99)
Dotfiles installation complete.
coder@env> sudo -u root stat /root/.dotfiles.log                                                                                 ~
  File: /root/.dotfiles.log
  Size: 488             Blocks: 8          IO Block: 4096   regular file
Device: 10008dh/1048717d        Inode: 266456      Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2024-03-07 17:29:59.822682570 +0000
Modify: 2024-03-07 17:29:48.174934810 +0000
Change: 2024-03-07 17:29:48.174934810 +0000
 Birth: 2024-03-07 17:29:47.694945204 +0000
```

Reply:
That looks fine, you could do `~"$DOTFILES_USER"` (vs `~$DOTFILES_USER`) just to be on the safe side.

Reply:
Since we have to escape anyways, I opted for single instead of double quotes just to be on the safer side.
[63c0712](https://github.com/phorcys420/coder-modules/commit/63c0712cd4a67a260b9e9aee07e83004a6839dee)